Mathlib/Data/Real/EReal.lean

```diff
@@ -1239,6 +1245,71 @@ instance : HasDistribNeg EReal where
     rw [x.mul_comm, x.mul_comm]
     exact y.neg_mul x
 
+theorem right_distrib_of_nneg {a b c : EReal} (ha : 0 ≤ a) (hb : 0 ≤ b) :
+    (a + b) * c = a * c + b * c := by
+  rcases eq_or_lt_of_le ha with (rfl | a_pos)
+  · simp
+  rcases eq_or_lt_of_le hb with (rfl | b_pos)
+  · simp
+  rcases lt_trichotomy c 0 with (c_neg | rfl | c_pos)
+  · induction c using EReal.rec
+    · rw [mul_bot_of_pos a_pos, mul_bot_of_pos b_pos, mul_bot_of_pos (add_pos a_pos b_pos),
+        add_bot ⊥]
+    · induction a using EReal.rec
+      · exfalso; exact not_lt_bot a_pos
+      · induction b using EReal.rec
+        · norm_cast
+        · norm_cast; exact right_distrib _ _ _
+        · norm_cast
+          rw [add_top_of_ne_bot (coe_ne_bot _), top_mul_of_neg c_neg, add_bot]
+      · rw [top_add_of_ne_bot (ne_bot_of_gt b_pos), top_mul_of_neg c_neg, bot_add]
+    · exfalso; exact not_top_lt c_neg
+  · simp
+  · induction c using EReal.rec
+    · exfalso; exact not_lt_bot c_pos
+    · induction a using EReal.rec
+      · exfalso; exact not_lt_bot a_pos
+      · induction b using EReal.rec
+        · norm_cast
+        · norm_cast; exact right_distrib _ _ _
+        · norm_cast
+          rw [add_top_of_ne_bot (coe_ne_bot _), top_mul_of_pos c_pos,
+            add_top_of_ne_bot (coe_ne_bot _)]
+      · rw [top_add_of_ne_bot (ne_bot_of_gt b_pos), top_mul_of_pos c_pos,
+          top_add_of_ne_bot (ne_bot_of_gt (mul_pos b_pos c_pos))]
+    · rw [mul_top_of_pos a_pos, mul_top_of_pos b_pos, mul_top_of_pos (add_pos a_pos b_pos),
+        top_add_top]
+
+theorem left_distrib_of_nneg {a b c : EReal} (ha : 0 ≤ a) (hb : 0 ≤ b) :
+    c * (a + b) = c * a + c * b := by
+  nth_rewrite 1 [EReal.mul_comm]; nth_rewrite 2 [EReal.mul_comm]; nth_rewrite 3 [EReal.mul_comm]
+  exact right_distrib_of_nneg ha hb
+
+theorem le_iff_le_forall_real_gt (x y : EReal) :
+    y ≤ x ↔ ∀ (z : ℝ), (x < z) → (y ≤ z) := by
```

Comment:
Unless there's precedent elsewhere, I think the simpler expression should be on the right hand side.

Also, please remove superfluous parentheses

Reply:
Yes, there are precedents: 
- https://github.com/leanprover-community/mathlib4/blob/58c10cb5f7fdbedeb99512536e1079bb59990dcb/Mathlib/Algebra/Ring/Defs.lean
- https://github.com/leanprover-community/mathlib4/blob/58c10cb5f7fdbedeb99512536e1079bb59990dcb/Mathlib/Data/ZMod/Defs.lean
- https://github.com/leanprover-community/mathlib4/blob/58c10cb5f7fdbedeb99512536e1079bb59990dcb/Mathlib/NumberTheory/LucasLehmer.lean
- https://github.com/leanprover-community/mathlib4/blob/58c10cb5f7fdbedeb99512536e1079bb59990dcb/Mathlib/RingTheory/OreLocalization/Ring.lean
- https://github.com/leanprover-community/mathlib4/blob/58c10cb5f7fdbedeb99512536e1079bb59990dcb/Mathlib/SetTheory/Game/Basic.lean

The parentheses in `c * (a + b) = c * a + c * b` don't seem to be superfluous.

Reply:
Ops, sorry. Your comments are perfectly fine, but they were displayed aside the wrong line on VS Code.

Reply:
Done. Thanks.